File: day_01/puzzle_1.py
from pprint import pprint
import math

def get_fuel_req(mass):
    return math.floor(mass / 3) - 2

with open("input.txt", "r") as ifh:
    lines = ifh.readlines()

masses = [int(x) for x in lines]
fuel_reqs = [get_fuel_req(m) for m in masses]

total_fuel = 0
for f in fuel_reqs:
    total_fuel += f

print("total fuel:", total_fuel)

File: go/day_01/puzzle_1.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
)

func check(err error, msg string) {
	if err != nil {
		panic(fmt.Sprintf("%s: %s", msg, err))
	}
}

func readIntsFromFile(path string) []int {

	f, err := os.Open(path)
	check(err, fmt.Sprintf("Problem opening file %s", path))
	defer f.Close()

	var data []int

	scanner := bufio.NewScanner(f)
	for scanner.Scan() {

		s := scanner.Text()
		i, err := strconv.Atoi(s)
		check(err, "Problem converting string to int")

		data = append(data, i)
	}
	if err := scanner.Err(); err != nil {
		fmt.Fprintln(os.Stderr, "reading standard input:", err)
	}

	return data

}

func calcRequiredFuel(mass int) int {
	return int(math.Floor(float64(mass)/3)) - 2
}

func main() {

	data := readIntsFromFile("input.txt")
	totalFuel := 0

	for _, mass := range data {
		fuel := calcRequiredFuel(mass)
		totalFuel += fuel
	}

	fmt.Printf("total fuel: %v\n", totalFuel)

}
